osu.Game/Rulesets/UI/FrameStabilityContainer.cs

```diff
@@ -85,35 +85,52 @@ protected override void LoadComplete()
 
         public override bool UpdateSubTree()
         {
-            state = frameStableClock.IsPaused.Value ? PlaybackState.NotValid : PlaybackState.Valid;
+            if (parentGameplayClock == null)
+                setClock(); // LoadComplete may not be run yet, but we still want the clock.
+
+            double aimTime = parentGameplayClock.CurrentTime;
+
+            if (frameStableClock.WaitingOnFrames.Value)
+            {
+                // waiting on frames is a special case where we want to avoid doing any update propagation, unless new frame data has arrived.
```

Comment:
What's the issue with the "update propagation" happening? All we're doing is updating the clocks right? Over which both updateReplay() and updateClock() have full control over the proposed time and can choose to set it arbitrarily...

Regardless of how that's done the final state should be NotValid and the update "propagation" won't happen anyway?

Reply:
I moved this out to avoid running the updateClock method while waiting on frames. By running it, it updates the manualClock's state which I wanted to avoid at all costs.

But I think you may be right in that we don't want to call SetFrameFromTime twice in a row, as it will advance two frames, potentially.

Reply:
> we don't want to call SetFrameFromTime twice in a row

That's the sort of stuff I'm concerned about. As another example, this also has no check for `ReplayInputHandler` being null, which is already checked prior to `updateReplay()`. Which is why I mentioned that it's hard to follow this logic above.

> updates the manualClock's state which I wanted to avoid at all costs

But why? What part of it would break? Couldn't this very same logic, placed inside updateReplay(), set `proposedTime = manualClock.CurrentTime` and forego such updates?